tests/HUBViewModelRendererTests.m

```diff
@@ -0,0 +1,126 @@
+ */
+@interface HUBCollectionViewMockWithoutBatchUpdates : HUBCollectionViewMock
+
+@end
+
+@implementation HUBCollectionViewMockWithoutBatchUpdates
+
+- (void)insertItemsAtIndexPaths:(NSArray<NSIndexPath *> *)indexPaths
+{
+}
+
+- (void)deleteItemsAtIndexPaths:(NSArray<NSIndexPath *> *)indexPaths
+{
+}
+
+- (void)reloadItemsAtIndexPaths:(NSArray<NSIndexPath *> *)indexPaths
+{
+}
+
+@end
+
+@interface HUBViewModelRendererTests : XCTestCase
+
+@property (nonatomic, strong) HUBCollectionViewMockWithoutBatchUpdates *collectionView;
+@property (nonatomic, strong) HUBCollectionViewLayoutMock *collectionViewLayout;
+@property (nonatomic, strong) HUBViewModelRenderer *viewModelRenderer;
+
+@end
+
+@implementation HUBViewModelRendererTests
+
+- (void)setUp
+{
+    [super setUp];
+
+    self.collectionView = [HUBCollectionViewMockWithoutBatchUpdates new];
+    self.collectionViewLayout = [[HUBCollectionViewLayoutMock alloc] init];
+    self.collectionView.collectionViewLayout = self.collectionViewLayout;
+    self.viewModelRenderer = [HUBViewModelRenderer new];
+}
+
+- (void)tearDown
+{
+    self.collectionView = nil;
+    self.collectionViewLayout = nil;
+    self.viewModelRenderer = nil;
+
+    [super tearDown];
+}
+
+- (void)testTwoSubsequentRenders
+{
+    NSArray<id<HUBComponentModel>> *firstComponents = @[
+        [HUBViewModelUtilities createComponentModelWithIdentifier:@"component-1" customData:nil],
+    ];
+    id<HUBViewModel> firstViewModel = [HUBViewModelUtilities createViewModelWithIdentifier:@"Test" components:firstComponents];
+
+    NSArray<id<HUBComponentModel>> *secondComponents = @[
+        [HUBViewModelUtilities createComponentModelWithIdentifier:@"component-2" customData:nil],
+    ];
+    id<HUBViewModel> secondViewModel = [HUBViewModelUtilities createViewModelWithIdentifier:@"Test2" components:secondComponents];
+
+    __weak XCTestExpectation * const expectation = [self expectationWithDescription:@"Waiting for render"];
+
+    [self.viewModelRenderer renderViewModel:firstViewModel inCollectionView:self.collectionView usingBatchUpdates:YES animated:YES addHeaderMargin:YES completion:^{
+        // Immediately trigger another render.
+        [self.viewModelRenderer renderViewModel:secondViewModel inCollectionView:self.collectionView usingBatchUpdates:YES animated:YES addHeaderMargin:YES completion:^{
+            [expectation fulfill];
+        }];
+    }];
+
+    [self waitForExpectationsWithTimeout:2 handler:nil];
+
+    XCTAssertEqual(self.collectionViewLayout.capturedViewModels.count, 2u);
+    XCTAssertEqualObjects(self.collectionViewLayout.capturedViewModels[0], firstViewModel);
+    XCTAssertEqualObjects(self.collectionViewLayout.capturedViewModels[1], secondViewModel);
+
+    XCTAssertEqual(self.collectionViewLayout.capturedViewModelDiffs.count, 2u);
+    // The first invocation shouldn't generate a diff.
+    XCTAssertEqualObjects(self.collectionViewLayout.capturedViewModelDiffs[0], [NSNull null]);
```

Comment:
@rastersize That's a good point, and most likely the reason why the builds are crashing on Travis. I'll fix it.